sygnal/http.py

```diff
@@ -218,13 +218,13 @@ def find_pushkins(self, appid):
         if appid in self.sygnal.pushkins:
             return [self.sygnal.pushkins[appid]]
 
-        result = []
-        for key, value in self.sygnal.pushkins.items():
-            # The ".+" symbol is used in place of "*" symbol
-            regex = key.replace("*", ".+")
-            if re.search(regex, appid):
-                result.append(value)
-        return result
+        # otherwise, find any pushkins whose appid patterns match
+        pushkins = []
+        for pushkin_appid, pushkin in self.sygnal.pushkins.items():
+            pattern = glob_to_regex(pushkin_appid, ignore_case=False)
```

Comment:
It seems like it would be nice to precompile these, but it's less easy than the other case so I guess it depends on how ugly you think that'd be. (It also short-circuits to doing a direct check in the dict for the exact app ID, so I don't know how commonly this case is hit.)

Reply:
I've changed the patterns to be explicitly cached on startup.